gef.py

```diff
@@ -7673,6 +7888,116 @@ def do_invoke(self, argv, *args, **kwargs):
 
         for title, content in data:
             gef_print("{}: {}".format(Color.boldify("{:<22}".format(title)), content))
+
+        ptype = {
+            Phdr.PT_NULL:         "NULL",
+            Phdr.PT_LOAD:         "LOAD",
+            Phdr.PT_DYNAMIC:      "DYNAMIC",
+            Phdr.PT_INTERP:       "INTERP",
+            Phdr.PT_NOTE:         "NOTE",
+            Phdr.PT_SHLIB:        "SHLIB",
+            Phdr.PT_PHDR:         "PHDR",
+            Phdr.PT_TLS:          "TLS",
+            Phdr.PT_LOOS:         "LOOS",
+            Phdr.PT_GNU_EH_FRAME: "GNU_EH_FLAME",
+            Phdr.PT_GNU_STACK:    "GNU_STACK",
+            Phdr.PT_GNU_RELRO:    "GNU_RELRO",
+            Phdr.PT_LOSUNW:       "LOSUNW",
+            Phdr.PT_SUNWBSS:      "SUNWBSS",
+            Phdr.PT_SUNWSTACK:    "SUNWSTACK",
+            Phdr.PT_HISUNW:       "HISUNW",
+            Phdr.PT_HIOS:         "HIOS",
+            Phdr.PT_LOPROC:       "LOPROC",
+            Phdr.PT_HIPROC:       "HIPROC",
+        }
+
+        pflags = {
+            0:                             "---",
+            Phdr.PF_X:                     "--X",
```

Comment:
There is a `Permission` class, you can use it too like:
```python
            Phdr.PF_X =                    Permission.EXECUTE,
```
then you can simply use `str()` on it.

And so on for the following

Reply:
Doesn't really work out, since that's just the enum (which would just be converted into a number).

We would have to convert it into a Permission class at some point by either doing something like

`Phdr.PF_W|Phdr.PF_X:           Permission(value=Permission.WRITE | Permission.EXECUTE),`

or

`Phdr.PF_R|Phdr.PF_X:           Permission.from_process_maps("r-x"),`

or otherwise (when using the enum only) would have convert it to permission class while printing

`str(Permission(value=p_flags))`

What do you prefer?

Reply:
I guess what I was trying to say is we can reuse existing code since we have from your PR

```python
    PF_X            = 1
    PF_W            = 2
    PF_R            = 4
```

and already existing:

```python
class Permission:
    NONE      = 0
    READ      = 1
    WRITE     = 2
    EXECUTE   = 4
```

Which could allow you to use the object `Permission` below in the `for` loop something
```python
for i, p in enumerate(elf.phdrs):
   [...]
   #  p_flags = pflags[p.p_flags]
   p_flags = Permission(value=p.p_flags)
```
Since the class defines `__str__` it will show the mask a string you'd expect